crypto/keys/internal/ecdsa/pubkey.go

```diff
@@ -46,9 +56,26 @@ func (pk *PubKey) Bytes() []byte {
 }
 
 // VerifySignature checks if sig is a valid ECDSA signature for msg.
+// This includes checking for low-s normalized signatures
+// where the s integer component of the signature is in the
+// lower half of the curve order
+// 7/21/21 - expects raw encoded signature (fixed-width 64-bytes, R || S)
+
 func (pk *PubKey) VerifySignature(msg []byte, sig []byte) bool {
-	s := new(signature)
-	if _, err := asn1.Unmarshal(sig, s); err != nil || s == nil {
+
+	// check length for raw signature
+	// which is two 32-byte padded big.Ints
+	// concatenated
+	// NOT DER!
+
+	if len(sig) != 64 {
+		return false
+	}
+
+	s := signatureFromBytes(sig)
+
+	// if not low-s, then fail validation
+	if !IsSNormalized(s.S) {
```

Comment:
it might be non-trivial, but is there any way we could add a test for `VerifySignature` with a non-normalized signature?

Reply:
Yes, I think that is a good thing to have. In general, there are a number of test vectors that could or should be added for the signature code. The low-s normalization one would be one of the easier ones to do (I think).

Reply:
I figured out how to add this test into the signature tests, by extracting the r and s values from the valid signature with the test key, scalar-negating s so it goes into the high curve order (the exact ECDSA "flaw") and then requiring validation to fail. Just added that commit. Resolving this conversation.